packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -754,20 +757,39 @@ export class MirrorNodeClient {
    * In some very rare cases the /contracts/results api is called before all the data is saved in
    * the mirror node DB and `transaction_index` or `block_number` is returned as `undefined` or `block_hash` as `0x`.
    * A single re-fetch is sufficient to resolve this problem.
-   * @param {string} transactionIdOrHash - The transaction ID or hash
-   * @param {RequestDetails} requestDetails - The request details for logging and tracking.
+   *
+   * @param {Function} getContractResultFunc - The function used to fetch the contract result.
+   *                                           It should accept parameters as a spread of arguments.
+   * @param {any[]} params - The parameters to be passed to `getContractResultFunc`.
+   *                         These are used to fetch the contract result.
+   * @returns {Promise<any>} - The resolved contract result, either fetched on the first attempt or after a retry.
    */
-  public async getContractResultWithRetry(transactionIdOrHash: string, requestDetails: RequestDetails) {
-    const contractResult = await this.getContractResult(transactionIdOrHash, requestDetails);
-    if (
-      contractResult &&
-      !(
-        contractResult.transaction_index &&
-        contractResult.block_number &&
-        contractResult.block_hash != EthImpl.emptyHex
-      )
-    ) {
-      return this.getContractResult(transactionIdOrHash, requestDetails);
+  public async getContractResultWithRetry(
+    getContractResultFunc: (...param: any) => any,
+    params: any[],
+    requestDetails: RequestDetails,
+  ): Promise<any> {
+    const shortDelay = 250;
+    const contractResult = await getContractResultFunc.call(this, ...params);
+    const contractObjects = Array.isArray(contractResult) ? contractResult : [contractResult];
+
+    for (const contractObject of contractObjects) {
+      if (
+        contractObject &&
+        (contractObject.transaction_index == null ||
+          contractObject.block_number == null ||
+          contractObject.block_hash == EthImpl.emptyHex)
+      ) {
+        if (this.logger.isLevelEnabled('debug')) {
+          this.logger.debug(
+            `${requestDetails.formattedRequestId} Contract result contains undefined transaction_index, block_number, or block_hash set to 0x. Retrying after a delay of ${shortDelay} ms.`,
```

Comment:
Might as well log the 3 fields as we won't know which field is null in this case

Reply:
You're right that would give more info. Added those fields and an extra field for transaction hash for easier debug.